test/cmd/version.sh

```diff
@@ -32,8 +32,8 @@ run_kubectl_version_tests() {
 
   # create version files, one for the client, one for the server.
   # these are the files we will use to ensure that the remainder output is correct
-  kube::test::version::object_to_file "Client" "" "${TEMP}/client_version_test"
-  kube::test::version::object_to_file "Server" "" "${TEMP}/server_version_test"
+  kube::test::version::json_client_server_object_to_file "" "clientVersion.gitVersion" "${TEMP}/client_version_test"
```

Comment:
`client_version_test` and `server_version_test` now just contain a version number, whereas before they contained all the various fields parsed out of the long version output.

What if you leave these two lines (35 and 36) unchanged and change lines 46-56, to use `.gitVersion` like this:
```
  kube::log::status "Testing kubectl version: verify json output"
  kube::test::version::json_client_server_object_to_file "" "clientVersion.gitVersion" "${TEMP}/client_json_version_test"
  kube::test::version::json_client_server_object_to_file "" "serverVersion.gitVersion" "${TEMP}/server_json_version_test"
  kube::test::version::diff_assert "${TEMP}/client_version_test" "eq" "${TEMP}/client_json_version_test" "--output json has correct client info"
  kube::test::version::diff_assert "${TEMP}/server_version_test" "eq" "${TEMP}/server_json_version_test" "--output json has correct server info"

  kube::log::status "Testing kubectl version: verify json output using additional --client flag does not contain serverVersion"
  kube::test::version::json_client_server_object_to_file "--client" "clientVersion.gitVersion" "${TEMP}/client_only_json_version_test"
  kube::test::version::json_client_server_object_to_file "--client" "serverVersion.gitVersion" "${TEMP}/server_client_only_json_version_test"
  kube::test::version::diff_assert "${TEMP}/client_version_test" "eq" "${TEMP}/client_only_json_version_test" "--client --output json has correct client info"
  kube::test::version::diff_assert "${TEMP}/server_version_test" "ne" "${TEMP}/server_client_only_json_version_test" "--client --output json has no server info"
```

Does that fix the integration tests?